lib/evaluation/inputDescriptorFilterEvaluationHandler.ts

```diff
@@ -0,0 +1,94 @@
+import jp from 'jsonpath';
+import Ajv from 'ajv';
+import { Field, InputDescriptor, PresentationDefinition } from '@sphereon/pe-models';
+import { Status } from '../ConstraintUtils';
+import { AbstractEvaluationHandler } from './abstractEvaluationHandler';
+import { HandlerCheckResult } from './handlerCheckResult';
+
+export class InputDescriptorFilterEvaluationHandler extends AbstractEvaluationHandler {
+
+
+    public getName(): string {
+        return 'FilterEvaluation';
+    }
+
+    public handle(pd: PresentationDefinition, p: any, results: HandlerCheckResult[]): void {
+        const inputDescriptors: InputDescriptor[] = pd.input_descriptors;
+        for (const vc of p.verifiableCredential.entries()) {
+            this.iterateOverInputDescriptors(inputDescriptors, vc, results);
+        }
+    }
+
+    private iterateOverInputDescriptors(inputDescriptors: InputDescriptor[], vc: [number, any], results: HandlerCheckResult[]): void {
+        for (const inputDescriptor of inputDescriptors.entries()) {
+            if (inputDescriptor[1].constraints && inputDescriptor[1].constraints.fields && inputDescriptor[1].constraints.fields.length > 0) {
```

Comment:
Extract method

this.hasFields(inputDescriptor)

private hasFields(inputDescriptor: [number, InputDescriptor]) {
    return inputDescriptor[1].constraints && inputDescriptor[1].constraints.fields && inputDescriptor[1].constraints.fields.length > 0;
  }

Reply:
Fixed!